external-crates/move/crates/move-analyzer/src/completion.rs

```diff
@@ -172,53 +172,33 @@ fn get_cursor_token(buffer: &str, position: &Position) -> Option<Tok> {
 /// Handle context-specific auto-completion requests with lbrace (`{`) trigger character.
 fn context_specific_lbrace(
     symbols: &Symbols,
-    use_fpath: &Path,
-    position: &Position,
-) -> Vec<CompletionItem> {
-    let mut completions = vec![];
-
-    // look for a struct definition on the line that contains `{`, check its abilities,
-    // and do auto-completion if `key` ability is present
-    for u in symbols.line_uses(use_fpath, position.line) {
-        let def_loc = u.def_loc();
-        let Some(use_file_mod_definition) = symbols.file_mods.get(use_fpath) else {
-            continue;
-        };
-        let Some(use_file_mod_def) = use_file_mod_definition.first() else {
-            continue;
-        };
-        if !is_definition(
-            symbols,
-            position.line,
-            u.col_start(),
-            use_file_mod_def.fhash(),
-            def_loc,
-        ) {
-            continue;
-        }
-        let Some(def_info) = symbols.def_info(&def_loc) else {
-            continue;
-        };
-        let DefInfo::Struct(.., abilities, _, _, _) = def_info else {
-            continue;
-        };
-        if abilities.has_ability_(Ability_::Key) {
-            let obj_snippet = "\n\tid: UID,\n\t$1\n".to_string();
-            let init_completion = CompletionItem {
-                label: "id: UID".to_string(),
-                kind: Some(CompletionItemKind::SNIPPET),
-                documentation: Some(Documentation::String("Object snippet".to_string())),
-                insert_text: Some(obj_snippet),
-                insert_text_format: Some(InsertTextFormat::SNIPPET),
-                ..Default::default()
-            };
-            completions.push(init_completion);
-            break;
+    cursor: &CursorContext,
```

Comment:
Thanks for that! I was thinking if it would be possible to do it with IDE annotations to make it robust wrt line breaks, but it makes a lot of sense with the cursor!